### ISSUE TITLE
Progress.. :)

### ISSUE THREAD

**Issue by powerpaul17** [XePeleato/android_device_huawei_p8lite_alel21#1]
Hello! Sorry for starting this as an issue but I can't find a way to write messages to users on Github (maybe it doesn't exist..)

I found your repository for the P8 Lite yesterday and since I am also trying to compile a AOKP ROM for this device I wanted to ask if your device configuration is working. I also saw that you already included the Mali libraries, is this working too. So far I couldn't get my phone to boot correctly..

Thanks and maybe we can combine our efforts..?

Greetings, Paul

**Comment by XePeleato** [comment]
Hi there, the build works, but the device won't boot to android due to some GPU issues that I am trying to fix.
As far as I can tell, the device configuration is fine, the current issue is with the Mali GPU, but the libs get loaded by the device.

Feel free to contribute, some help would be awesome!

**Comment by concept0** [comment]
Hello! I am very glad that you are working on CM for alel21. Can the USB Host / OTG support for him? Unfortunately I'm still a noob and cant do it myself. Sincerely, Alexander.

**Comment by powerpaul17** [comment]
Hey, thanks for the reply. I also have the same problem with the GPU, it seems to boot but the graphics do not get initialized.

Did you use a stock kernel? Maybe it has something to do with this, I am currently trying to compile the kernel too. I also wrote to someone on xda-developers who seems to have accomplished to compile the kernel.

**Comment by XePeleato** [comment]
Well, first I am trying to get working pure Android, whenever that's done, I'll start with Cm.

**Comment by XePeleato** [comment]
@powerpaul17 The kernel gets compiled from source, I am pushing that commit right now.

**Comment by XePeleato** [comment]
@powerpaul17 I'd love to see your logcat output, mine is this

http://pastebin.com/iRMw1SBS

Is surfaceflinger also crashing on your end?

**Comment by powerpaul17** [comment]
sorry, I haven't saved the logcat, but it looks pretty much the same.. it also crashed at the same point with the same stacktrace. I also tried to use the surfaceflinger binary from the stock rom but then it doesn't even get started.

I start to believe it might have something to do with the kernel. Maybe the stock kernel is not compatible with AOSP, I am trying to compile the kernel from the sources but I still got some issues.

**Comment by XePeleato** [comment]
I think I've reached a major milestone :P
![dsc00164](https://cloud.githubusercontent.com/assets/6912584/12874750/933cacbc-cdda-11e5-9476-dbaa7f33f748.JPG)

**Comment by powerpaul17** [comment]
Nice.. :) So is it booting? Did you manage to compile the kernel? I am still trying..

**Comment by XePeleato** [comment]
@powerpaul17 It's almost booting. What's that kernel issue? Are you using Linaro toolchain?

**Comment by powerpaul17** [comment]
I am using the toolchain which gets compiled or prepared from the "repo" system (just started with Android development.. ;) ). The kernel finished compiling now using the sources from Huawei but I haven't tested yet. What are your boot issues?

**Comment by XePeleato** [comment]
@powerpaul17 zygote is crashing, at least surfaceflinger loads properly, it looks like I'll need to change some libs.

**Comment by olrak2** [comment]
Any progress ?

thx in advance.

**Comment by XePeleato** [comment]
@olrak2 Well, I have compiled the kernel succesfully, and now the "rom" itself is my obejctive, if there's any major progress I'll post it here or wherever I find it convenient.

Have a nice day!

**Comment by XePeleato** [comment]
Just a quick update for those of you who are waiting, I have spent all of the weekend working on this, I have compiled more than 40 images so far just in the weekend, I have solved some issues but the rom is not finished yet. Now I am not stuck with the graphics system anymore, but there are more issues to fix, I won't surrender.

**Comment by olrak2** [comment]
Good luck with the problems :) I'm very glad you won't surrender.  We need a rom for our Little P8

**Comment by XePeleato** [comment]
Alright, got it, the rom boots to android, I am not too excited about it since some services crash and the rom is not functional, since you are spammed with "service XXX stopped working" warnings, but at least, it boots :) I'll upload some pics later

**Comment by olrak2** [comment]
I'm very excited to hear that. Those warning's can't be worse than EMUI :) Let those pics come :)

**Comment by XePeleato** [comment]
![screenshot_2016-03-07-20-00-26 1](https://cloud.githubusercontent.com/assets/6912584/13582909/13179588-e4af-11e5-9430-6da1fc42119a.png)

![screenshot_2016-03-07-20-00-21 1](https://cloud.githubusercontent.com/assets/6912584/13582927/23ef819a-e4af-11e5-98f5-677c68781b3f.png)

There are lots of bugs (as expected), but there's the proof!

**Comment by ghost** [comment]
wow, congrats, so it was libs issue?

**Comment by XePeleato** [comment]
Kind of @hak86 I also disabled selinux and modified the ramdisk, I'll submit a pull request in the afternoon to your tree, since I have spotted something weird.

**Comment by ghost** [comment]
@XePeleato i think you need to remove everything used by emui, second we should make a ril.java file to let it works, we can try with honor 6 one, it's use kirin920 if not wrong but i'm not sure ;)

**Comment by XePeleato** [comment]
@hak86 True, since the phone doesn't even ask me to unlock the SIM card. I didn't had much luck following the Honor 6 config. I removed every Huawei file inside 'framework', so there's no trace of Huawei code (just the libs used to get the camera/wifi/bluetooth/display..) working.

**Comment by ghost** [comment]
one question, does get in the out like root/res? needed for the charger? @XePeleato because my own boot won't boot without them....pretty weird...

**Comment by XePeleato** [comment]
@hak86 Yes, I copied them from the original ramdisk, it's not updated here because I am building everything again from scratch.

**Comment by XePeleato** [comment]
UPDATE: I am switching from AOSP to CM, because AOSP doesn't let me configure some things that CM does, and at the moment, I don't have the time to figure out how to do them on my own, so the Build machine is now compiling, allow some hours for a new update.

Have a nice weekend!

**Comment by XePeleato** [comment]
Well, it looks like it wasn't as easy as just switching sources, now it doesn't even boot.

**Comment by ghost** [comment]
i fix copying blobs into build and now should copy them with any rom, for the boot.img i still don't understand why in the out/target/product/hi6210sft/root, give me only sbin and res, should be folder like data, dev, system and many other..bah.. @XePeleato

**Comment by XePeleato** [comment]
Well, you  can copy them by hand in order to check if it works @hak86  Mine copied that files just fine.

**Comment by olrak2** [comment]
@XePeleato Can you update your source?  I also want to test aosp and try to boot up my P8 lite. One thing i find weird.  How does it come your build number is the same as the original stock build number? It should be a total different number than the stock one, no?

**Comment by XePeleato** [comment]
@olrak2  It doesn't work that way, the build number is written into a file, I uses the file that came with the device, so it isn't so surprising

**Comment by olrak2** [comment]
Good to know :) I don't know anything about building.  Still learning.  Is your tree up to date so i can test on it?

**Comment by XePeleato** [comment]
@olrak2 Kind of, however you can download it and test it, but let me remind you that it isn't even finished, and you'll need to use my kernel in order to get it working, also, folders "vendor" and "etc" are not copied, since I am doing it by hand because I am testing those files at the moment, However you can still copy them yourself

**Comment by XePeleato** [comment]
I'm getting closer :)

![screenshot_2016-03-23-13-00-46](https://cloud.githubusercontent.com/assets/6912584/13985979/1f410f76-f100-11e5-9ebc-b6487268d167.png)

**Comment by ghost** [comment]
nice, i just finish a build too, let's see :+1:

**Comment by XePeleato** [comment]
Update. 

I have been talking to codeworkx and it looks like the whole RIL stack isn't compatible with the AOSP framework. As I see it, We  have three options. 
1. We can decompile Huawei's framework 
2. We can develop a hack
3. We can beg Huawei

Any Java developer is up to a challenge?

**Comment by ghost** [comment]
did you try to call the command like adb shell > rild to see if drivers are missed?

**Comment by XePeleato** [comment]
Yes, however it worked. 'adb logcat - b radio' was  more explicit

**Comment by muhammadfahadbaig** [comment]
@XePeleato What about the patching the modemst images.

**Comment by muhammadfahadbaig** [comment]
@XePeleato Remove the third option. **3. We can beg Huawei** I have done that already and too much...

**Comment by muhammadfahadbaig** [comment]
@XePeleato which sources you used to compile RIL

**Comment by XePeleato** [comment]
The RIL daemon is the original one, and the libs are also from Huawei. If  you mean the Java class, it's just used to fix signal strength and some other minor bugs.

**Comment by muhammadfahadbaig** [comment]
@XePeleato Talking about codeworks, he did compiled CM 11 for Honor 6. Any tip from him? I am seriously not good in Java (even I am below moderate in C/C#). You mentioned, decompiling Huawei Framework. Any idea from where to start? I know someone who may take a look.

**Comment by olrak2** [comment]
@XePeleato Why not try omnirom or cyanogenmod again?

Probably also a good idea is to update your source tree so everybody can build from your source.  

I gave tried to build but could not boot to android.

There were a couple of weird things in your tree that i needed to fix before i could build:
- BoardConfig.mk line 90 needted to remove: TW_THEME := hdpi_portrait
- BoardConfig.mk TARGET_KERNEL_PREBUILT  does not exist it's : TARGET_PREBUILT_KERNEL
- You import in aosp_hi6210sft.mk as well in device.mk your ramdisk (only needed once)

Could you please also upload your vendor files so we can use them.  I have seen a dir /vendor/huawei/p8litekirin/ and a dir /vendor.  Where did you place the vendor files?

Here was my catlog file from my bootup: http://pastebin.com/raw/F1744gqG

Many thanks for the work that you already done. :smiley:

**Comment by XePeleato** [comment]
The tree is updated, I'll upload the vendor files in the evening. Thanks!

**Comment by olrak2** [comment]
No, thank you @XePeleato

**Comment by sgspluss** [comment]
@XePeleato would it be helpfull to take a look into huawei p6? They fix ril problems, maybe this will helps you to fix it to. Just search "huawei p6 ril fix" in google.

**Comment by XePeleato** [comment]
@sgspluss Since it doesn't support Lollipop, I don't think it'll work.

However thank you for your suggestion!

**Comment by sgspluss** [comment]
I know it didnt has lollipop but maybe it will be helpful anyway @XePeleato, hope you can fix it _finger-cross_

**Comment by sgspluss** [comment]
@XePeleato, just search a bit around and found following things, all for CM12.1...

https://github.com/ShevT/vendor_huawei_U9500/commit/0de7be416167dab5ac53677cfdbff1044347e4be

https://github.com/ShevT/device_huawei_U9500/commit/625ed001a380605584de526c3cd006cdc90c737d#

http://forum.xda-developers.com/showpost.php?p=62770987&postcount=7 

The last link sounds that the ril fix for CM11 should work bur just for single sim, dual sim have problems.
Well however hope that links can help to find the problem for our device...

**Comment by muhammadfahadbaig** [comment]
@sgspluss It still will be difficult. Apart from the RIL blobs, the QMI blobs won't work for our device. QMI is a protocol developed by Qualcomn and is only available on Qualcomn Chipsets.
Our best bet would be to,
1. Patch the Modem Images to make them compatible.
2. Switch to CM base.
3. Or maybe give that QMI fix a try :) Atleast nothing bad would happen.

**Comment by sgspluss** [comment]
Maybe in this case it would be the best to test it with cm base at first. Right it could be that the QMI fix doesn't work due to chipset, so we'll have to find an other solution sadly...

**Comment by muhammadfahadbaig** [comment]
@XePeleato did you checked the kernel source code. There might be some modem or RIL configurations. Another bet could be to edit them and compile the modified modules and make them compatible with AOSP.

**Comment by muhammadfahadbaig** [comment]
@XePeleato Since you switched to Android 5.1.1. Does it boots (I believe it'll boot fine)? Secondly, which release version you are using? There are so many releases r1 to r25 to r.... Does picking right release matters?

**Comment by XePeleato** [comment]
It boots x I am using r1

**Comment by VenityZ** [comment]
@XePeleato First of hello , I have been watching this project for a while and I see you make a lot of progress. Nicely done. I would like to try it out if it's ok for you. Could you tell me everything I need in order to build it? The newest Kernel Source of you I guess? And which vendor?

**Comment by muhammadfahadbaig** [comment]
@VenityZ You don't need Kernel Sources. A prebuilt kernel is included with this repository. Secondly, you have to extract vendor files yourself from your device. At the moment, a script to automatically extract the needed vendor files is not included. Therefore, see `aosp_hi6210sft.mk` to get a list of the vendor files needed and where they have to be placed.

**Comment by XePeleato** [comment]
I forgot to upload the blobs, since you can get them from your device, I'll post them whenever I can.

**Comment by VenityZ** [comment]
@muhammadfahadbaig Thanks. Saw the Kernel wasn't sure about that. A little Challenge for me.

**Comment by XePeleato** [comment]
Since I can't seem to get RIL working, I guess I'll post the Alpha version of the rom in order to build up interest. A true shame.

**Comment by muhammadfahadbaig** [comment]
@XePeleato Don't say "True Shame". The problem with Huawei is the lack of community of developers who are willing to spare time and devote in development. You are a lone ranger who took the interest in development and got this far. Now that's a true achievement. Also, you're the first one to create a bootable 5.1.1 ROM for Huawei HiSilicon chipsets. We need people to gain interest in this project and maybe someone who has enough experience shines some light in this regard.
However, even if you have decided to upload the ROM, make sure to warn people about the ROM's current state. You might not want a flood of comments saying A is crashing, B doesn't opens, etc.

**Comment by XePeleato** [comment]
Well, not really, the help I received here was essential to develop the rom, I am a C developer and I have stuff to work on and I have never developed an android rom before, so I just used  the office Build server and my spare time. I don't really want to release the rom, because it isn't finished but we need  as much help as possible. 

Now, I have discovered that hwframework (I don't remember  the full name of it) invokes RIL and manages it, that framework is developed by Huawei, Will CM fix this? I really doubt it, however we'll know pretty soon. Anyway it looks like we'll have to patch our framework, and I don't even know where to start.

**Comment by muhammadfahadbaig** [comment]
CM won't fix this but CM is relatively easier in fixing bugs (thanks to it's large community support) but very difficult in getting it to boot.

**Comment by VenityZ** [comment]
Does the RIL effect 3g and etc. I don't know if I can help but you need someone who has Java knowledge. I have mush but not in things such this , gonna have a look and try to find someone who knows what to do.

**Comment by muhammadfahadbaig** [comment]
The first step would be to deodex the framework files. Due to Android ART Runtime, the framework classes are stored in a boot.oat file. So the actual framework files contain nothing useful as most of the classes are pulled away and stored in a separate file. I'll edit this post as soon as I get the deodexed framework.

**Comment by muhammadfahadbaig** [comment]
@VenityZ RIL (Radio Interface Layer) manages the complete baseband and modem. Simply, your cellular network. It is a means of transport or whatever else between the software and hardware.

**Comment by VenityZ** [comment]
@muhammadfahadbaig Already knew this but thx anyway. On my old device it had a more important role. 
@XePeleato If you need help , I could make a public post with a full explanation of the project and with its problems where  I search for people to help. If you give me a yes I will make one.

**Comment by muhammadfahadbaig** [comment]
@XePeleato Did you decompiled the hwframework.jar? I would upload it right now if's not.

**Comment by muhammadfahadbaig** [comment]
@XePeleato in **aosp_hi6210sft.mk** add these libraries `/vendor/lib/libreference-ril.so`, `/vendor/lib64/libreference-ril.so`, `/vendor/lib/librilutils.so` and `/vendor/lib64/librilutils.so`.
Also I've got 3 other Anonymous developers to look into this RIL problem :)

**Comment by XePeleato** [comment]
I really appreciate your help,thank you! But unfortunately it doesn't work

**Comment by sgspluss** [comment]
Could it be have someting to do with this?

Stock:
-#BOARD_RIL_CLASS := ../../../device/HUAWEI/hi6210sft/ril/

Edit in yours:
+BOARD_RIL_CLASS := src/java/com/android/internal/telephony

Maybe not vaild path or else?

**Comment by XePeleato** [comment]
@sgspluss That's because I have moved it,however that RIL class just fixes the signal strength and other minor bugs, so even if it's not loading correctly, it's not causing the problem.

**Comment by sgspluss** [comment]
@XePeleato thanks for clearify this. Then will hold my eyes open again for this ril problem. Dont have much experiances with rom (just few small one) more with kernel, but anyway try my best to help out whenever i can. ;-)

**Comment by muhammadfahadbaig** [comment]
If anyone here knows anything about Java development, I have uploaded the disassembled hwframework.jar here: https://drive.google.com/file/d/0B0LG0AQ4V9FpN0xmZERsbEdVQTg/view?usp=sharing. The file uploaded is a .zip file. Inside it are .smali files which can be edited using any text editor (such as Notepad++).
This framework can be used as a reference to patch AOSP framework (which unfortunately I don't know how to)

**Comment by olrak2** [comment]
@XePeleato , you already done a very good job, you are my hero :smile: 

But could you please tell us how you were able to build?  I tried to build from your tree, it worked after adjusting some settings but got an error when i booted.  adb error.  Maybe a little manual :wink: 

Never been able to get a full boot. Always stuck on boot.

I would like to help with the ril problem but if i can't make a build it's not easy

**Comment by muhammadfahadbaig** [comment]
@olrak which Android repo you're using?

**Comment by olrak2** [comment]
@muhammadfahadbaig , i have tried all of them.  Android 5.0, Android 5.1 R1, etc...

I always get this error : http://pastebin.com/raw/F1744gqG

**Comment by muhammadfahadbaig** [comment]
@olrak2
1. Downgrade back to Android 5.1.1 (if you are on Android 6.0) i.e. flash stock Huawei Android 5.1.1.
2. The stock boot.img won't work. Netd (Network Daemon) is failing maybe because @XePeleato made some changes to kernel (NETFILTERS???) and init.rc in ramdisk. Compile a boot.img yourself using this repository kernel and ramdisk files.

**Comment by VenityZ** [comment]
@muhammadfahadbaig of which vesion is the Framework?

**Comment by muhammadfahadbaig** [comment]
@VenityZ It's stock Huawei Framework Android 5.1.1. EMUI 3.1

**Comment by VenityZ** [comment]
@muhammadfahadbaig I don't know if it changes something. But you were using B008 I guess theres a newer 5.1.1. I think the frameworks are different need update pakage?
Also are the permissions in the ueventd.rc right not familiar the modem.

**Comment by muhammadfahadbaig** [comment]
@VenityZ Actually, we don't need any EMUI framework patches. We only need how hwframework invokes and manages RIL. You may compare all RIL classes in my uploaded version to any other Android 5.1.1 EMUI version. I think it would be the same.

**Comment by XePeleato** [comment]
@muhammadfahadbaig Nice job! Let me play around with it, and let's see if I can understand it.

**Comment by ghost** [comment]
i manage to got worked boot.img with adb too compiled under cm-12.1, and also i have an idea want to apply after see first boot to get worked ril without patches, keep in touch :+1:

**Comment by muhammadfahadbaig** [comment]
@hak86 @XePeleato All fighting for one common cause.... to free everyone from the horribly designed (although with many great features and addons) EMUI.

**Comment by ghost** [comment]
@muhammadfahadbaig sure bro, our device will be faster with a stable rom i think :)

**Comment by muhammadfahadbaig** [comment]
@olrak2 So were you able to get it to boot?

**Comment by muhammadfahadbaig** [comment]
@XePeleato About the RIL issue, I was able to find RIL libraries for HiKey boards. And HiKey Board is compatible with AOSP. Any chance they might work?

**Comment by XePeleato** [comment]
@muhammadfahadbaig No idea, where did you find them? I could try.

**Comment by muhammadfahadbaig** [comment]
@XePeleato Here is the link for complete AOSP ROM for HiKey: https://builds.96boards.org/snapshots/hikey/linaro/aosp/218/ . Download system.img.tar.xz. Also for a list of libs and other files included in this distribution, see installed-files.txt
I think the libs would work. GPS, Bluetooth, and maybe RIL, I hope they'll work.

**Comment by muhammadfahadbaig** [comment]
@XePeleato I built from your sources. Unfortunately, my device is stuck on Huawei logo. It doesn't even goes to bootscreen. Any idea why? (using the boot.img compiled from sources)

**Comment by muhammadfahadbaig** [comment]
@olrak2 @XePeleato I confirm the bug faced by @olrak2 . Stuck at Net.d starting....
http://pastebin.com/eyVhnnAE

**Comment by olrak2** [comment]
@muhammadfahadbaig, after looking in the system.img from @XePeleato , you need to copy the entire bin, etc, lib, lib64 dir's to the out directory of your build. You need to put it all in the system folder.  After copying the dir's you can build your rom. 

Tomorrow i post a scrypt that you can put in your Android.mk file.

**Comment by muhammadfahadbaig** [comment]
@olrak2 Yeah, I figured that out. The reason of Net.d is crashing was either because of init.rc in ramdisk or because of the Android Runtime crashing. The latter is caused because of incompatible libraries being used in the ROM which are not compatible with AOSP.
@XePeleato It would be great if you upload the vendor files you're using. It seems that some vendor files which I'm using are not compatible with AOSP and hence the Android Runtime crashing. It would be great if you upload the vendor files to ensure compatibility and that anyone who is trying to build from your source atleast gets a bootable build.

**Comment by XePeleato** [comment]
It's not a good idea  to copy the whole folders, I'll upload it this afternoon

**Comment by muhammadfahadbaig** [comment]
@XePeleato Right about that! I'm using Android 5.1.1 r37 and I would not like to face some major crashes. Also, it's good you uploaded the 5.1.1 system.img on XDA. I'll extract the libs from that and then I'll create a repository here on github.

**Comment by muhammadfahadbaig** [comment]
@XePeleato @olrak2 Here are the vendor files :) https://github.com/muhammadfahadbaig/vendor_device_hi6210sft

**Comment by XePeleato** [comment]
Development is officially resumed!

**Comment by olrak2** [comment]
@XePeleato Good news

If you need something, just call.  

My working tree for omnirom 5.1 is on my github. (https://github.com/olrak2/android_device_huawei_alice)

**Comment by FelipeBR28** [comment]
How's everything going? And what about the RIL?

**Comment by XePeleato** [comment]
There's still work to be done, the RIL stack won't work until I patch the framework.

**Comment by ghost** [comment]
and also does required patches to boot the rom? my own aosp still stuck due netd is failing to work..

**Comment by XePeleato** [comment]
@hak86 Are We talking about Lollipop or Marshmallow?

**Comment by ghost** [comment]
at the moment lollipop, mm downloads, in progress.

**Comment by XePeleato** [comment]
@hak86 Not really, I didn't even touched the framework at Lollipop

**Comment by ghost** [comment]
so with a correct blobs will boot and this drivers like gatekeeper.hi6210sft.so was use on emui 4.0, right? i am trying your tree right now, once booted will try my modified ril .java and see if will work.

**Comment by XePeleato** [comment]
@hak86 My tree can be useful as a start point but You'll have to tweak it a little bit

**Comment by ghost** [comment]
@XePeleato yep i see, there was some libs are missed, i am sure it's not a problem with ramdisk, my own ramdisk is complete and worked with omni, in just 4 files. will report if got moved (still building).

**Comment by theel0ja** [comment]
Some progress?
